src/main/java/net/fabricmc/mappingio/tree/VisitOrder.java

```diff
@@ -264,11 +268,125 @@ public boolean isMethodVarsFirst() {
 		return methodVarsFirst;
 	}
 
+	private static final AlphanumComparator alphanumComparator = new AlphanumComparator();
 	private Comparator<ClassMappingView> classComparator;
 	private Comparator<FieldMappingView> fieldComparator;
 	private Comparator<MethodMappingView> methodComparator;
 	private Comparator<MethodArgMappingView> methodArgComparator;
 	private Comparator<MethodVarMappingView> methodVarComparator;
 	private boolean methodsFirst;
 	private boolean methodVarsFirst;
+
+	/*
+	 * The Alphanum Algorithm is an improved sorting algorithm for strings
+	 * containing numbers.  Instead of sorting numbers in ASCII order like
+	 * a standard sort, this algorithm sorts numbers in numeric order.
+	 *
+	 * The Alphanum Algorithm is discussed at http://www.DaveKoelle.com
+	 *
+	 * Released under the MIT License - https://opensource.org/licenses/MIT
```

Comment:
Hmm, I'm not a lawyer but doesn't this mean that a copy of the MIT license needs to be included with all copies of mio, including compiled jars?

Reply:
Oh it might do, I wrongly assumed MIO was MIT 🤦  We need to check this before merging, I assume you looked hard for an existing sort algo? I feel like this might already exist somewhere.

Reply:
I think it would be best to put this in its own file, maybe even package with a license file next to it and in the header.

Reply:
> Hmm, I'm not a lawyer but doesn't this mean that a copy of the MIT license needs to be included with all copies of mio, including compiled jars?
Tbf I have no idea what the existing conventions/regulations are, I assumed it would be fine since we're distributing the sources JAR with the comment included anyway.

> I assume you looked hard for an existing sort algo?
It's surprisingly hard to find one, this one here was recommended by multiple people online and didn't appear to have any bugs, in comparison to some alternatives.

> I think it would be best to put this in its own file, maybe even package with a license file next to it and in the header.
Which package name do you recommend? `com.davekoelle.alphanum`?

Reply:
Done!